module/VuFindConsole/tests/unit-tests/src/VuFindTest/Command/Util/IndexReservesCommandTest.php

```diff
@@ -330,9 +350,260 @@ public function testSuccessWithILS()
         $commandTester = new CommandTester($command);
         $commandTester->execute([]);
         $this->assertEquals(0, $commandTester->getStatusCode());
-        $this->assertEquals(
-            "Successfully loaded 3 rows.\n",
+        $this->assertStringContainsString(
+            'Successfully loaded 3 rows.',
+            $commandTester->getDisplay()
+        );
+    }
+
+    /**
+     * Test successful ILS loading with some invalid data in reserves.
+     *
+     * @return void
+     */
+    public function testSuccessWithILSWithInvalidData()
+    {
+        $ils = $this->getMockIlsConnection();
```

Comment:
I wonder if it's worth creating a convenience method like `getMockIlsConnectionWithCourseData` to set up the consecutive call expectation so you don't have to duplicate that long logic in so many places (and so it will be easy to fix the test if some future refactoring should somehow change the order of operations and require adjustments). Not a requirement, but just a suggestion in case you like the idea!

Reply:
Good idea. I made a new helper function and have it returning the `CommandTester` object so the individual test functions could make calls like `$this->assertEquals(1, $commandTester->getStatusCode());` afterwards. I debated putting the `$commandTester->execute([]);` line in the function too but my initial reaction was it seemed odd to have a function that just created an object actually run some tests.

Reply:
Agreed, I think this makes sense -- thanks for taking the time!